app/src/main/java/org/schabi/newpipe/util/external_communication/KoreUtils.java

```diff
@@ -29,13 +33,39 @@ public static boolean shouldShowPlayWithKodi(@NonNull final Context context,
                 .getBoolean(context.getString(R.string.show_play_with_kodi_key), false);
     }
 
-    public static void showInstallKoreDialog(@NonNull final Context context) {
-        final AlertDialog.Builder builder = new AlertDialog.Builder(context);
-        builder.setMessage(R.string.kore_not_found)
-                .setPositiveButton(R.string.install, (dialog, which) ->
-                        NavigationHelper.installKore(context))
-                .setNegativeButton(R.string.cancel, (dialog, which) -> {
-                });
-        builder.create().show();
+    /**
+     * Start an activity to install Kore.
+     *
+     * @param context the context
+     */
+    public static void installKore(final Context context) {
+        installApp(context, context.getString(R.string.kore_package));
+    }
+
+    /**
+     * Start Kore app to show a video on Kodi, and if the app is not installed ask the user to
+     * install it.
+     * <p>
+     * For a list of supported urls see the
+     * <a href="https://github.com/xbmc/Kore/blob/master/app/src/main/AndroidManifest.xml">
+     * Kore source code
+     * </a>.
+     *
+     * @param context  the context to use
+     * @param videoURL the url to the video
+     */
+    public static void playWithKore(final Context context, final Uri videoURL) {
```

Comment:
You should keep camel case style for method parameters.

Also, Kore supports SoundCloud links, so `videoUrl` should be renamed to something like `streamUrl`:

```suggestion
    public static void playWithKore(final Context context, final Uri streamUrl) {
```

Reply:
I just moved this code, but you're right